cpp/src/reductions/legacy/group_std.cu

```diff
@@ -0,0 +1,201 @@
+    T* d_sums = sums.data().get();
+
+    auto values_it = thrust::make_transform_iterator(
+      thrust::make_counting_iterator(0),
+      [=] __device__ (cudf::size_type i) {
+        if (values_valid and not bit_mask::is_valid(values_valid, i))
+          return 0.0;
+        
+        double x = values_data[i];
+        cudf::size_type group_idx = d_group_labels[i];
+        cudf::size_type group_size = d_group_sizes[group_idx];
+        
+        // prevent divide by zero error
+        if (group_size == 0 or group_size - ddof <= 0)
+          return 0.0;
+
+        double mean = static_cast<double>(d_sums[group_idx])/group_size;
+        return (x - mean) * (x - mean) / (group_size - ddof);
+      }
+    );
+
+    thrust::reduce_by_key(rmm::exec_policy(stream)->on(stream),
+                          group_labels.begin(), group_labels.end(), values_it, 
+                          thrust::make_discard_iterator(),
+                          result_data);
+
+    // set nulls
+    if (result_valid) {
+      thrust::for_each_n(rmm::exec_policy(stream)->on(stream),
+        thrust::make_counting_iterator(0), group_sizes.size(),
+        [=] __device__ (cudf::size_type i){
+          cudf::size_type group_size = d_group_sizes[i];
+          if (group_size == 0 or group_size - ddof <= 0)
+            bit_mask::clear_bit_safe(result_valid, i);
+          else
+            bit_mask::set_bit_safe(result_valid, i);
+        });
+      set_null_count(*result);
+    }
+
+    // if std, do a sqrt
+    if (is_std) {
+      thrust::transform(rmm::exec_policy(stream)->on(stream),
+                        result_data, result_data + group_sizes.size(),
+                        result_data,
+        [] __device__ (double data) { return sqrt(data); });
+    }
+  }
+
+  template <typename T, typename... Args>
+  std::enable_if_t<!std::is_arithmetic<T>::value, void >
+  operator()(Args&&... args) {
+    CUDF_FAIL("Only numeric types are supported in variance");
+  }
+};
+
+} // namespace anonymous
+
+
+namespace cudf {
+namespace detail {
+
+void group_var(gdf_column const& values,
+               rmm::device_vector<size_type> const& group_labels,
+               rmm::device_vector<size_type> const& group_sizes,
+               gdf_column * result,
+               size_type ddof,
+               cudaStream_t stream)
+{
+  type_dispatcher(values.dtype, var_functor{},
+    values, group_labels, group_sizes, result, ddof, false, stream);
+}
+
+void group_std(gdf_column const& values,
+               rmm::device_vector<size_type> const& group_labels,
+               rmm::device_vector<size_type> const& group_sizes,
+               gdf_column * result,
+               size_type ddof,
+               cudaStream_t stream)
+{
+  type_dispatcher(values.dtype, var_functor{},
+    values, group_labels, group_sizes, result, ddof, true, stream);
+}
+
+} // namespace detail
+
+std::pair<cudf::table, cudf::table>
+group_std(cudf::table const& keys,
```

Comment:
I think there should only be detail APIs for this, since it should be accessed via a groupby operator just like variance.

Reply:
See other comment. Standalone-only for now. Integrated, later.